tools/launch.ts

```diff
@@ -171,11 +174,62 @@ function start() {
 
   const portPrefix = argv["port-prefix"] || 34;
   const startingPort = portPrefix * 1000;
-  const parachainName = argv.parachain.toString();
-  const parachain = parachains[parachainName];
-  const parachainChain = argv["parachain-chain"] || parachain.chain;
+  let paras = [];
+  let parasNames = [];
+  let parachainsChains = [];
+  let paraIds = [];
+
+  // We start gathering all the information about the parachains
+  if (Array.isArray(argv["parachain-id"])) {
+    // We need two validators per parachain, so there is a maximum we can support
+    if (argv["parachain-id"].length * 2 > validatorNames.length) {
+      console.error(`Exceeded max number of paras: ${validatorNames.length / 2}`);
+      return;
+    }
```

Comment:
I hadn't heard the two-validators-per-parachain rule before. Did you observe that yourself, or is it written somewhere?

(I meant to post this comment days ago, but must have gotten distracted)

Reply:
When I had only two, polkadot-launch told me some like "You need at least as many validators as parachains you have". But I could only make it work if I assigned two validators per parachain. Something to take a further look in the future definitely